source/slang/ir-legalize-types.cpp

```diff
@@ -343,6 +350,9 @@ static LegalVal legalizeFieldExtract(
 {
     auto builder = context->builder;
 
+    if (type.flavor == LegalType::Flavor::none)
```

Comment:
Same as in `legalizeFieldAddress`, we need to check whether the field we are loading is an empty struct. If it is, just return None.